pkg/workflows/wasm/host/module.go

```diff
@@ -393,39 +432,56 @@ func containsCode(err error, code int) bool {
 	return strings.Contains(err.Error(), fmt.Sprintf("exit status %d", code))
 }
 
-func fetchFn(logger logger.Logger, modCfg *ModuleConfig) func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
-	const fetchErrSfx = "error calling fetch"
+func createFetchFn(
+	logger logger.Logger,
+	reader unsafeReaderFunc,
+	writer unsafeWriterFunc,
+	sizeWriter unsafeFixedLengthWriterFunc,
+	modCfg *ModuleConfig, store *store,
+) func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
+	logFetchErr := func(err error) { logger.Errorf("error calling fetch: %s", err.Error()) }
 	return func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
-		b, innerErr := wasmRead(caller, reqptr, reqptrlen)
+		b, innerErr := reader(caller, reqptr, reqptrlen)
 		if innerErr != nil {
-			logger.Errorf("%s: %s", fetchErrSfx, innerErr)
+			logFetchErr(innerErr)
 			return ErrnoFault
 		}
 
 		req := &wasmpb.FetchRequest{}
 		innerErr = proto.Unmarshal(b, req)
 		if innerErr != nil {
-			logger.Errorf("%s: %s", fetchErrSfx, innerErr)
+			logFetchErr(innerErr)
+			return ErrnoFault
+		}
+
+		storedRequest, innerErr := store.get(req.Id)
+		if innerErr != nil {
+			logFetchErr(innerErr)
+			return ErrnoFault
+		}
+
+		if storedRequest.ctxFn() == nil {
+			logFetchErr(errors.New("context is nil"))
 			return ErrnoFault
 		}
 
-		fetchResp, innerErr := modCfg.Fetch(req)
+		fetchResp, innerErr := modCfg.Fetch(storedRequest.ctxFn(), req)
```

Comment:
If this is too messy:
```suggestion
		var fetchResp Whatever
		var innerErr error 
		storedRequest.callWithCtx(func(ctx context.Context) {
			fetchResp, innerErr = modCfg.Fetch(ctx, req)
		})
```
Then the signature can be expanded to match the return args:
```suggestion
		fetchResp, innerErr := storedRequest.callWithCtx(func(ctx context.Context) (Whatever, error) {
			return modCfg.Fetch(ctx, req)
		})
```